src/Symfony/Component/Uuid/Uuid.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uuid;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class Uuid implements \JsonSerializable
+{
+    public const TYPE_1 = UUID_TYPE_TIME;
+    public const TYPE_3 = UUID_TYPE_MD5;
+    public const TYPE_4 = UUID_TYPE_RANDOM;
+    public const TYPE_5 = UUID_TYPE_SHA1;
+
+    public const VARIANT_NCS = UUID_VARIANT_NCS;
+    public const VARIANT_DCE = UUID_VARIANT_DCE;
+    public const VARIANT_MICROSOFT = UUID_VARIANT_MICROSOFT;
+    public const VARIANT_OTHER = UUID_VARIANT_OTHER;
+
+    private $uuid;
+
+    public function __construct(string $uuid)
+    {
+        if (!uuid_is_valid($uuid)) {
```

Comment:
we might miss a `public static isValid(): bool` method

Reply:
I added it to make you happy but I think it's totally useless:
```
    public static function isValid(string $uuid): bool
    {
        return uuid_is_valid($uuid);
    }
```

Reply:
I agree with Greg, why should we leak state here. I mean, one could/should instantiate this object in a try/catch ? 🤔

Reply:
All the class is "just a proxy".
We need to think about ppl that won't use the Value Object but will use strings instead. Forcing them to use a try/catch is something they won't like, DX wise.